jetstream/backgroundtaskmessage-notification-release-4pct-de.toml

```diff
@@ -0,0 +1,194 @@
+        '{{experiment.start_date_str}}' AND
+        '{{experiment.last_enrollment_date_str}}'
+    -- The background update experiment slug is exact.
+    AND experiment.key = '{{experiment.normandy_slug}}'
+GROUP BY client_id, branch
+)
+
+UNION ALL
+
+(
+SELECT
+    client_id AS client_id,
+    -- Post [Bug 1804988](https://bugzilla.mozilla.org/show_bug.cgi?id=1804988),
+    -- this name looks like 'slug:branch'.
+    SPLIT(mozfun.map.get_key(event_map_values, 'name'), ':')[SAFE_OFFSET(1)] AS branch,
+    MIN(submission_date) AS enrollment_date,
+    COUNT(submission_date) AS num_enrollment_events
+FROM
+    `mozdata.telemetry.events`
+WHERE
+    -- Browsing telemetry should not be delayed, but notification clicks need
+    -- not coincide with actual enrollment.
+    submission_date BETWEEN
+        '{{experiment.start_date_str}}' AND
+        '{{experiment.last_enrollment_date_str}}'
+    AND event_category = 'browser.launched_to_handle'
+    AND event_method = 'system_notification'
+    AND event_object = 'toast'
+    -- Post [Bug 1804988](https://bugzilla.mozilla.org/show_bug.cgi?id=1804988),
+    -- this name looks like 'slug:branch'.
+    AND STARTS_WITH(mozfun.map.get_key(event_map_values, 'name'), '{{experiment.normandy_slug}}:')
+GROUP BY
+    client_id, branch
+)
+
+)
+QUALIFY ROW_NUMBER() OVER (PARTITION BY client_id ORDER BY enrollment_date ASC) = 1
+"""
+
+[data_sources.browser_launched_events]
+from_expression = """(
+    SELECT
+        *
+    FROM `mozdata.telemetry.events`
+    WHERE event_category = 'browser.launched_to_handle'
+)"""
+
+# These events will not be paired with an 'experiments' annotation, so don't
+# expect it -- the annotation is on the background update data, not the browsing
+# data queried here.
+experiments_column_type = "none"
+friendly_name = "Browser Launched Events"
+description = "Browser Launched Events"
+
+[metrics]
+# Even though some enrollment telemetry can be quite delayed, the click
+# telemetry should not be delayed, and therefore calculating sums and rates
+# daily will help understand if the experiment is incorrectly configured.
+daily = ["notification_clicks", "active_hours", "search_count", "days_of_use", "qualified_cumulative_days_of_use"]
+weekly = ["notification_clicks"]
+overall = ["notification_clicks"]
+
+[metrics.notification_clicks]
+# Exposure events are not yet reliable: see [Bug 1809275](https://bugzilla.mozilla.org/show_bug.cgi?id=1809275).
+analysis_bases = ["enrollments"]
+data_source = "browser_launched_events"
+select_expression = '''{{agg_any(
+    """
+    event_method = 'system_notification'
+    AND event_object = 'toast'
+    -- After https://github.com/mozilla/jetstream/issues/1516 is addressed,
+    -- we can handle this generically using the experiment slug; but right
+    -- now, we need to interpolate the specific slug by hand.
+    AND STARTS_WITH(mozfun.map.get_key(event_map_values, 'name'), 'backgroundtaskmessage-notification-release-4pct-de:')
```

Comment:
> Each experiment is copy-paste with a single slug updated. https://github.com/mozilla/jetstream/issues/1516 would allow to avoid that, but it's had no comment since I filed it.

I commented on [jetstream#1516](https://github.com/mozilla/jetstream/issues/1516) that I think that functionality does exist. In theory, we'd be able to change this line to (something like)

`AND mozfun.map.get_key(event_map_values, 'name') = '{experiment_slug}`

Reply:
If that's the case, we can factor the metric definition out to, and reference it in each experiment. Though exactly where we should factor it to isn't clear to me. Perhaps it should go in `/definitions/firefox_desktop.toml`, since that seems to be where the other metric definitions are being pulled from (even though this is a background updater experiment)